plugins/package-managers/node/src/main/kotlin/Yarn.kt

```diff
@@ -87,14 +87,12 @@ class Yarn(
 
         val process = run(workingDir, "info", "--json", packageName)
 
-        return Json.parseToJsonElement(process.stdout).jsonObject["data"]?.let {
+        val data = Json.parseToJsonElement(process.stdout).jsonObject["data"]?.also {
             yarnInfoCache.write(packageName, it.toString())
-            parsePackageJson(it)
-        } ?: run {
-            Json.parseToJsonElement(process.stderr).jsonObject["data"].let {
-                logger.warn { "Error running '${process.commandLine}' in directory $workingDir: $it" }
-                parsePackageJson(checkNotNull(it))
-            }
+        } ?: checkNotNull(Json.parseToJsonElement(process.stderr).jsonObject["data"]).also {
+            logger.warn { "Error running '${process.commandLine}' in directory $workingDir: $it" }
         }
+
+        return parsePackageJson(data)
```

Comment:
That's funny. I almost had a change exactly like this, but kept it out of the PR as I wasn't sure about the `checkNotNull`. But as you now come up with this as well, let's just do this. ^^